raui-core/src/widget/mod.rs

```diff
@@ -591,6 +652,29 @@ macro_rules! destruct {
     };
 }
 
+/// A helper for getting the named children out of a widget context
+///
+/// # Example
+///
+/// ```
+/// # use raui_core::prelude::*;
+/// fn my_component(context: WidgetContext) -> WidgetNode {
+///     // Destructure our context to get our named slots
+///     let WidgetContext {
+///         named_slots,
+///         ..
+///     } = context;
+///     // Unpack our named `body` slot
+///     unpack_named_slots!(named_slots => body);
```

Comment:
we should also add a snippet with multiple slots unpacking: `unpack_named_slots!(named_slots => { header, body, footer });`